packages/expo-image/android/src/main/java/expo/modules/image/ExpoImageView.kt

```diff
@@ -0,0 +1,188 @@
+package expo.modules.image
+
+import android.annotation.SuppressLint
+import android.graphics.BitmapFactory
+import android.graphics.Canvas
+import android.graphics.PorterDuff
+import android.graphics.drawable.Drawable
+import androidx.appcompat.widget.AppCompatImageView
+import com.bumptech.glide.RequestManager
+import com.bumptech.glide.load.model.GlideUrl
+import com.bumptech.glide.request.RequestOptions
+import com.facebook.react.bridge.ReactContext
+import com.facebook.react.bridge.ReadableMap
+import com.facebook.react.modules.i18nmanager.I18nUtil
+import com.facebook.react.uimanager.PixelUtil
+import com.facebook.react.uimanager.events.RCTEventEmitter
+import com.facebook.yoga.YogaConstants
+import expo.modules.image.drawing.BorderDrawable
+import expo.modules.image.drawing.OutlineProvider
+import expo.modules.image.enums.ImageResizeMode
+import expo.modules.image.events.ImageLoadEventsManager
+import expo.modules.image.okhttp.OkHttpClientProgressInterceptor
+
+private const val SOURCE_URI_KEY = "uri"
+private const val SOURCE_WIDTH_KEY = "width"
+private const val SOURCE_HEIGHT_KEY = "height"
+private const val SOURCE_SCALE_KEY = "scale"
+
+@SuppressLint("ViewConstructor")
+class ExpoImageView(context: ReactContext, private val requestManager: RequestManager, private val progressInterceptor: OkHttpClientProgressInterceptor) : AppCompatImageView(context) {
+  private val eventEmitter = context.getJSModule(RCTEventEmitter::class.java)
+  private val outlineProvider = OutlineProvider(context)
+  private var borderDrawable: BorderDrawable? = null
+  private var loadedSource: GlideUrl? = null
+  internal var sourceMap: ReadableMap? = null
+
+  init {
+    clipToOutline = true
+    scaleType = ImageResizeMode.COVER.scaleType
+    super.setOutlineProvider(outlineProvider)
+  }
+
+  internal fun setResizeMode(resizeMode: ImageResizeMode) {
+    scaleType = resizeMode.scaleType
+    // TODO: repeat mode handling
+  }
+
+  internal fun setBorderRadius(position: Int, borderRadius: Float) {
+    var radius = borderRadius
+    val isInvalidated = outlineProvider.setBorderRadius(radius, position)
+    if (isInvalidated) {
+      invalidateOutline()
+      if (!outlineProvider.hasEqualCorners()) {
+        invalidate()
+      }
+    }
+
+    // Setting the border-radius doesn't necessarily mean that a border
+    // should to be drawn. Only update the border-drawable when needed.
+    if (borderDrawable != null) {
+      radius = if (!YogaConstants.isUndefined(radius)) PixelUtil.toPixelFromDIP(radius) else radius
+      if (position == 0) {
+        borderDrawable!!.setRadius(radius)
+      } else {
+        borderDrawable!!.setRadius(radius, position - 1)
+      }
+    }
```

Comment:
```suggestion
	val borderDrawable = this.borderDrawable;
    if (borderDrawable != null) {
      radius = if (!YogaConstants.isUndefined(radius)) PixelUtil.toPixelFromDIP(radius) else radius
      if (position == 0) {
        borderDrawable.setRadius(radius)
      } else {
        borderDrawable.setRadius(radius, position - 1)
      }
    }
```
or 
```suggestion
	borderDrawable?.let {
	    if (borderDrawable != null) {
          radius = if (!YogaConstants.isUndefined(radius)) PixelUtil.toPixelFromDIP(radius) else radius
        if (position == 0) {
          it.setRadius(radius)
        } else {
          it.setRadius(radius, position - 1)
        }
    }
```
I don't like `!!` operator 😜